docs/DID Communications Research and Specification/i_did-introduction.md

```diff
@@ -0,0 +1,105 @@
+# did-introduction
+
+Describes how a go-between can introduce two parties that it already knows, but that do not know each other.
+
+### Roles
+- <u>**Introducer**</u>: Agent who introduces multiple <u>introducee</u>s to each other
+- <u>**Introducee**</u>: Two agents that are introduced to each other through the <u>introducer</u>
+
+Note that we have two roles here, but three agents.
+
+The ´comment´ field can be used to provide a rationale for the introduction, however, that only matters for humans, not for e.g. IoT devices. A rationale for devices would need to be standardized.
+
+When one of the <u>introducee</u>s denies the proposal and the other accepts, then the accepting party will know about the other party having denied the request, since it will not be followed through. This is akin to somebody denying a social media friend request and might pose somewhat of a privacy risk.
+
+### Flow
+
+![did-introduction flow](./img/did-introduction.png)
+
+### Messages
+
+#### introductionProposal
+The <u>introducer</u> sends the `introductionProposal` to every <u>introducee</u>, asking for their consent to the introduction.
+
+###### Layout
+
+```JSON
+introductionProposal: {
+    "context", // REQUIRED!
+    "thread", // REQUIRED!
+    "callbackURL", // REQUIRED!
+    "responseRequested", //OPTIONAL!
+    "id", // OPTIONAL!
+    "comment", // OPTIONAL!
+    "timing" // OPTIONAL!
+}
+```
+
+###### Example(s)
+
+```JSON
+{
+    "context": "did-introduction/1.0/introductionProposal",
+    "thread": "936DA01F9ABD4d9d80C702AF85C822A8",
+    "callbackURL": "https://www.bobsworld.com/"
+}
+```
+
+#### introductionResponse
+The <u>introducee</u>s answer with a `introductionResponse`, signaling their consent.
```

Comment:
It doesn't actually respond with yes or no. Since messages can be missed an explicit false response (no consent) would also be appreciated. So I think we need a boolean value response if consent is provided or not.

Reply:
Depends, you decide:
1) Either we model consent like aries does in e.g. [here](https://github.com/hyperledger/aries-rfcs/tree/master/features/0028-introduce), where we include an explicit boolean == new field into the interaction, or
2) We model consent like Jolocom does in [here](https://jolocom.github.io/jolocom-sdk/1.0.0/guides/interaction_flows/#authorization) by just repeating the message. I believe the rationale here is that nobody can expect an answer from you, so no answer == no

Reply:
Done 1) in 02527cc3